### PR TITLE
Updated readme

### PR DESCRIPTION
**Pull request by g4s8**
#31 - updated readme, added more details about `Slice`, `Response` and `Connection`.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#33](https://github.com/artipie/http/issues/33) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/548f302a-0aa9-431a-adb1-a3041324eddf); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: README.md [hunk 1]
```diff
@@ -26,6 +26,38 @@ This module tends to be reactive and provides these interfaces:
...
+  }
+}
+```
+
+Response is reactive object whith single method `send`. This method is called by
+server implementation, server provides connection implementation as `send` parameter
+which can accept response data: the server asks response to send itself to connection.
+
+```java
+class MavenResponse implements Slice {
```
Comment (fanifieiev) [inline README.md]: @g4s8 This class should implement [Response](https://github.com/artipie/http/blob/master/src/main/java/com/artipie/http/Response.java)

Comment (g4s8) [reply README.md]: @fanifiev fixed, thanks: https://github.com/artipie/http/pull/33/commits/49259ad9a969a3fdf5d256f731585811b082f4e0

File: README.md [hunk 2]
```diff
@@ -26,6 +26,38 @@ This module tends to be reactive and provides these interfaces:
...
+  }
+}
+```
+
+Response is reactive object whith single method `send`. This method is called by
+server implementation, server provides connection implementation as `send` parameter
+which can accept response data: the server asks response to send itself to connection.
+
+```java
+class MavenResponse implements Response {
```
Comment (fanifieiev) [inline README.md]: @g4s8 Please move the opening bracket to the upper line.

File: README.md [hunk 3]
```diff
@@ -26,6 +26,38 @@ This module tends to be reactive and provides these interfaces:
...
+
+    @Override
+    void send(final Connection con) {
+        con.accept(200, headers, empty);
```
Comment (fanifieiev) [inline README.md]: @g4s8 Please move the 'empty' parameter to the previous line.

File: README.md [hunk 4]
```diff
@@ -26,6 +26,38 @@ This module tends to be reactive and provides these interfaces:
  - `Connection` - response asks connection to accept response data, `Connection`
  should be implemented by HTTP web server implementation to accept HTTP responses
 
+Each artipie adapter has to implement `Slice` interface with single method `response`.
+This method should process the request and return reactive response object:
+```java
+class Maven implements Slice {
+  @Override
+  public Response response(String line, Iterable<Map.Entry<String, String>> headers,
```
Comment (fanifieiev) [inline README.md]: @g4s8 please  add 'final' keyword to the method parameters.

Comment (Sammers21) [reply README.md]: It is an example. The example should be as compact and understandable as possible.

Comment (fanifieiev) [reply README.md]: @Sammers21 No problem

**Comment by 0crat** [comment]
Job was finished in 20 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)[/z](https://www.0crat.com/u/fanifieiev) didn't make CR comments

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Pull request merged**